### PR TITLE
fix(vitest): fix `optimizeDeps.disabled` warnings on Vite 5.1

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

- closes https://github.com/vitest-dev/vitest/issues/5211
- related https://github.com/vitejs/vite/pull/15184

_todo_

- [x] figure out why warning shows up only when `customLogger` (see [comment](https://github.com/vitest-dev/vitest/issues/5211#issuecomment-1947810971))
- [x] add test based on https://github.com/vitest-dev/vitest/issues/4910
- [x] locally test Vite 5.1 + customLogger

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: test/optimize-deps/test/ssr.test.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @vitest-environment node
+
+import { expect, test } from 'vitest'
+
+// @ts-expect-error untyped
+import { importMetaUrl } from '@vitest/test-dep-url'
+
+test('import.meta.url', () => {
+  expect(importMetaUrl).toContain('/node_modules/.vitest/deps_ssr/')
```
Comment (hi-ogawa) [inline test/optimize-deps/test/ssr.test.ts:9]: Interestingly, ssr.optimizeDeps seems to be failing only on Windows.
Unfortunately `AssertionError` diff doesn't print everything...

Comment (sheremet-va) [reply test/optimize-deps/test/ssr.test.ts:9]: Shouldn't this be `\\node_modules\\.vitest\\deps_ssr`

Comment (hi-ogawa) [reply test/optimize-deps/test/ssr.test.ts:9]: It's `import.meta.url`, so I think path is already normalized with `file://something/node_modules/...`. The diff is truncated but you can see the first part here https://github.com/vitest-dev/vitest/actions/runs/7927404820/job/21643736718?pr=5215#step:8:697

Comment (hi-ogawa) [reply test/optimize-deps/test/ssr.test.ts:9]: Hmm, the same test passed on next run (though Windows failed on a different test) https://github.com/vitest-dev/vitest/actions/runs/7927585760/job/21644224823?pr=5215, so maybe this is the flakiness of optimizeDeps.

I'll retry a few times to see what happens.

Comment (sheremet-va) [reply test/optimize-deps/test/ssr.test.ts:9]: The flakiness is the reason why we disabled optimizer for now

Comment (hi-ogawa) [reply test/optimize-deps/test/ssr.test.ts:11]: It looks like only windows + ssr combination is flaky, so I only targeted this to disable.
Later I'll look into further if there's something sketchy with other cases.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**